File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.cs
public class Solution {
    public long MaximumHappinessSum(int[] happiness, int k) {
        Array.Sort(happiness);

        long result = 0;
        int decrease = 0;
        int n = happiness.Length;

        // Pick from largest to smallest
        for (int i = n - 1; i >= n - k; i--)
        {
            int current = happiness[i] - decrease;
            if (current > 0)
            {
                result += current;
            }
            decrease++;
        }

        return result;
    }
}

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.ts
function maximumHappinessSum(happiness: number[], k: number): number {
     happiness.sort((a, b) => a - b);

    let res = 0;
    let n = happiness.length;

    for (let i = n - 1, picked = 0; picked < k; i--, picked++) {
        const val = happiness[i] - picked;
        if (val > 0) res += val;
    }

    return res;
};